packages/plugin-trpc/cli/tests/lib/generate.spec.ts

```diff
@@ -0,0 +1,397 @@
+	const field: Field = {
+		type: 'field',
+		fieldType: 'Decimal',
+		name: 'decimalField'
+	};
+
+	const mappedType = mapZodType(field);
+	assert.is(mappedType, 'z.number()');
+});
+
+test('should import and extend the approuter', async () => {
+	const project = new Project({ useInMemoryFileSystem: true });
+
+	const sourceFile: SourceFile = project.createSourceFile(
+		'/testRouter.ts',
+		`
+		import { router, publicProcedure, mergeRouters } from './trpc';
+
+		const defaultRouter = router({
+		greeting: publicProcedure.query(() => 'hello webstone tRPC')
+		});
+
+		export const appRouter = mergeRouters(defaultRouter);
+
+		export type AppRouter = typeof appRouter;
+
+	`
+	);
+
+	prepareApprouter(sourceFile, 'testRouter', 'testRouter');
+
+	const imports = sourceFile.getImportDeclarations();
+
+	const namedImports = imports.flatMap((imp) =>
+		imp.getNamedImports().map((namedImp) => namedImp.getText())
+	);
+
+	const moduleSpecifiers = imports.map((imp) => imp.getModuleSpecifierValue());
+
+	assert.is(imports.length, 2);
+	assert.is(namedImports.length, 4);
+
+	// check named imports
+	assert.ok(namedImports.includes('testRouter'));
+
+	// check module specifiers
+	assert.ok(moduleSpecifiers.includes('./subrouters/testRouter'));
+});
+
+test('should return string as id type', async () => {
+	const model: Model = {
+		name: 'TestModel',
+		type: 'model',
+		properties: [
+			{
+				type: 'field',
+				fieldType: 'String',
+				name: 'stringField',
+				attributes: [
+					{
+						name: 'id',
+						type: 'attribute',
+						kind: 'field'
+					}
+				]
+			},
+			{
+				type: 'field',
+				fieldType: 'Non-Scalar',
+				name: 'nonScalarField'
+			},
+			{
+				type: 'field',
+				fieldType: 'Int',
+				name: 'numberField'
+			}
+		]
+	};
+
+	const idType = getIDType(model);
+	assert.is(idType, 'z.string()');
+});
+
+test('should return number as id type', async () => {
+	const model: Model = {
+		name: 'TestModel',
+		type: 'model',
+		properties: [
+			{
+				type: 'field',
+				fieldType: 'Int',
+				name: 'stringField',
+				attributes: [
+					{
+						name: 'id',
+						type: 'attribute',
+						kind: 'field'
+					}
+				]
+			},
+			{
+				type: 'field',
+				fieldType: 'Non-Scalar',
+				name: 'nonScalarField'
+			},
+			{
+				type: 'field',
+				fieldType: 'Int',
+				name: 'numberField'
+			}
+		]
+	};
+
+	const idType = getIDType(model);
+	assert.is(idType, 'z.number().int()');
+});
+
+test('should return error, because no ID provided', async () => {
+	try {
+		const model: Model = {
+			name: 'TestModel',
+			type: 'model',
+			properties: [
+				{
+					type: 'field',
+					fieldType: 'Int',
+					name: 'stringField'
+				},
+				{
+					type: 'field',
+					fieldType: 'Non-Scalar',
+					name: 'nonScalarField'
+				},
+				{
+					type: 'field',
+					fieldType: 'Int',
+					name: 'numberField'
+				}
+			]
+		};
+		getIDType(model);
+	} catch (error) {
+		assert.is(error.message, 'ID field not found');
+	}
+});
```

Comment:
Not all functions are tested properly just yet, we might want to do that before releasing, or do we want to add tests afterwards as well?

Reply:
I'd say let's not hold up the release of this because not all functions are tested.